src/@context/Asset.tsx

```diff
@@ -124,8 +124,8 @@ function AssetProvider({
 
     const accessDetails = await getAccessDetails(
       asset.chainId,
-      asset.services[0].datatokenAddress,
-      asset.services[0].timeout,
+      asset.services?.[0]?.datatokenAddress,
```

Comment:
This method should not fire when there are no services. So instead of all this optional chaining on here, the first line should be modified to return early. 

Most likely:

```js

if (!asset?.chainId || !asset?.services?.length) return
```